x-pack/plugins/security_solution/cypress/test_plans/detection_response/prebuilt_rules/prebuilt_rules_install_update_workflows.mdx

```diff
@@ -0,0 +1,190 @@
+**AND** installed rules are removed from Elastic Rules table  
+**AND** rules to install counter is decreased accordingly  
+
+*CASE 1: `<amount>` = All*  
+*CASE 2: `<amount>` = Selected*  
+
+#### **Scenario: User performing a clean install for prebuilt rules sees a loading skeleton until installation is completed**
+
+**GIVEN** a user that is on Rules Management Page  
+**WHEN** user installs all prebuilt rules through Add Elastic Rules button/link  
+**THEN** a loading skeleton is displayed until the installation is completed
+
+
+
+
+
+### Prebuilt Rules Update
+
+#### **Scenario: Users can update prebuilt rules**
+
+**GIVEN** user already has 8.x prebuilt rules installed in Kibana  
+**AND** there are new updates available for those prebuilt rules  
+**AND** user is on Rules Management Page  
+**WHEN** user navigates to the Rules Update tab  
+**THEN** user should see all the prebuilt rules  
+**AND** user can update outdated prebuilt rules  
+**AND** successfully updated message is displayed  
+**AND** Rules Upgrade tab counter is decreased according to the number of updated rules  
+
+
+
+
+### Installation / Update Failure
+
+#### **Scenario: Error message is displayed when any prebuilt rules operation fails**
+
+**GIVEN** user is `<action>` prebuilt rules    
+**WHEN** the installation or update process fails  
+**THEN** user should see an error message    
+**AND** prebuilt rules are not installed/updated    
+**AND** the callout message for pending installs/updates is still displayed on Rules Management Page  
+
+*CASE 1: `<action>` = installing all*  
+*CASE 2: `<action>` = installing selected*  
+*CASE 3: `<action>` = Updating selected*
+
+
+
+### Add Elastic Rules Page
+
+#### **Scenario: New workflow elements are displayed on Rules Management Page**
+
+**GIVEN** a user that doesn’t have `security_detection_engine` package installed  
+**WHEN** user is on Rules Management Page  
+**THEN** “+Add Elastic rules” menu with available Rules counter is displayed
+**AND** Rule Updates tab is displayed
+**AND** “+Add Elastic rules” button is displayed on empty Rules Table
+
+#### **Scenario: Rules settings persist on Add Elastic Rules table**
+
+**GIVEN** a user has Rules listed on Add Elastic Rules page  
+**WHEN** user reloads the page  
+**THEN** the rule state should persist for all the rules  
+*CASE 1: after refreshing the table*  
+*CASE 2: after switching table pagination*  
+*CASE 3: After filtering and clear filters*
+
+#### **Scenario: User can navigate back to Rules Management page**
+
+**GIVEN** a user is on Add Rules Page  
+**WHEN** user navigates back to Rules Management page  
+**THEN** Rules Management Page is properly displayed
+
+#### **Scenario: User can filter prebuilt rules by query or by tag**
+
+**GIVEN** a user is on Add Rules Page  
+**WHEN** user filters by `<filter>`  
+**THEN** Add Rules Table is properly updated
+
+*CASE 1: `<filter>` = Query filter on search bar*  
+*CASE 2: `<filter>` = Tag filter*
+
+
+
```

Comment:
Should we have a section analogous to the above (`### Add Elastic Rules Page`) but for the Rule Updates tab and table?

All scenarios except for the first one apply as well.

Reply:
Synced with @vgomez-el:

Yes, agreed. Let's also hear feedback from @banderror and @xcrzx if this section is necessary.

Reply:
These scenarios have been reworked. We have two empty sections to work on later:

```
### Rule installation workflow: filtering, sorting, pagination

TODO: add scenarios

### Rule upgrade workflow: filtering, sorting, pagination

TODO: add scenarios
```